scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # [self.t[k], self.t[-k]].
+        if extrapolate == 'periodic':
+            x = self.t[self.k] + (x - self.t[self.k]) % (self.t[-(self.k)] -
```

Comment:
It got quite confusing. You are correct that eventually we need to use `n = self.t.shape[0] - self.k - 1` (this line is from the constructor). 

I would suggest to store `n` explicitly in the class, i.e. assign `self.n = self.t.shape[0] - self.k - 1` in the constructor and then use it instead of recomputing. I believe @ev-br wouldn't mind such approach.